rmgpy/rmg/react.py

```diff
@@ -53,118 +54,98 @@ def react(*spcTuples):
+    # This method chops the iterable into a number of chunks which it
+    # submits to the process pool as separate tasks.
+    if procnum == 1:
+        logging.info('For reaction generation {0} process is used.'.format(procnum))
+        reactions = map(_react_species_star, spc_tuples)
+    else:
+        logging.info('For reaction generation {0} processes are used.'.format(procnum))
+        p = Pool(processes=procnum)
+        reactions = p.map(_react_species_star, spc_tuples)
+        p.close()
+        p.join()
 
-    results = map_(
-                reactSpecies,
-                spcTuples)
+    return itertools.chain.from_iterable(reactions)
 
-    reactions = itertools.chain.from_iterable(results)
 
-    return reactions
+def _react_species_star(args):
+    """Wrapper to unpack zipped arguments for use with map"""
+    return react_species(*args)
 
 
-def reactSpecies(speciesTuple):
+def react_species(species_tuple, only_families=None):
     """
     Given a tuple of Species objects, generates all possible reactions
     from the loaded reaction families and combines degenerate reactions.
-
-    The generated reactions are deflated.
     """
-    speciesTuple = tuple([spc.copy(deep=True) for spc in speciesTuple])
 
-    reactions = getDB('kinetics').generate_reactions_from_families(speciesTuple)
+    species_tuple = tuple([spc.copy(deep=True) for spc in species_tuple])
 
-    deflate(reactions,
-            [spec for spec in speciesTuple],
-            [spec.index for spec in speciesTuple])
+    reactions = getDB('kinetics').generate_reactions_from_families(species_tuple, only_families=only_families)
 
     return reactions
 
 
-def deflate(rxns, species, reactantIndices):
-    """
-    The purpose of this function is to replace the reactants and
-    products of a reaction, stored as Molecule objects by 
-    integer indices, corresponding to the species core index.
-
-    Creates a dictionary with Molecule objects as keys and newly 
-    created Species objects as values.
-
-    It iterates over the reactantIndices array, with elements in this array
-    corresponding to the indices of the core species. It creates a 
-    Molecule -> index entry in the previously created dictionary.
-
-    It iterates over the reaction list, and iteratively updates the
-    created dictionary as more reactions are processed.    
-    """    
-
-    molDict = {}
-
-    for i, coreIndex in enumerate(reactantIndices):
-        if coreIndex != -1:
-            for mol in species[i].molecule:
-                molDict[mol] = coreIndex
-
-    for rxn in rxns:
-        deflateReaction(rxn, molDict)
-        try:
-            deflateReaction(rxn.reverse, molDict) 
-        except AttributeError:
-            pass
-
-
-def reactAll(coreSpcList, numOldCoreSpecies, unimolecularReact, bimolecularReact, trimolecularReact=None):
+def react_all(core_spc_list, numOldCoreSpecies, unimolecularReact, bimolecularReact, procnum=1, trimolecularReact=None):
```

Comment:
I didn't notice this before, but I think it would be better to move `procnum` to be the last argument.

Reply:
Added fixup to "Moving determine_procnum_from_RAM() to rmgpy.rmg.main."